src/Symfony/Component/Messenger/MessageRecorderInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+interface MessageRecorderInterface extends ResetInterface
```

Comment:
SimpleBus split this interface into 2 small ones. The first one `ContainsRecordedMessages` is implement by our models because you need to tell them to give us the events raise during the transaction. The second one `RecordsMessages` is used to record event at this end of the transaction. For instance, I use it in my repository when my aggregate is flushed. I guess this two distinct responsibilities: I don't want that my model have a public method `record`.

Reply:
BTW, I prefer the naming of simple bus (declarative vs imperative programming). It is better to describer what happens instead how it happens :)

Reply:
I thought about slitting them. But I was not sure. Thank you for commenting!